src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        string += this[i];
+        break;
+      }
+
+      if (this[i] == null || typeof this[i] === 'undefined') {
+        string += separator;
+      } else {
+        string += this[i] + String(separator);
```

Comment:
```suggestion
        string += this[i] + separator;
```
i think separator is already a string

Reply:
if I`ll remove this, it won`t be work with null as a separator